### PR TITLE
Feature/naming and small issues

### PR DESCRIPTION
**Pull request by o-tsaruk**
Renamed all `_mint` to `_mint_to`;
Added `PSP22Capped` extension;
Added `_switch_pause` for `PSP22Pausable`;
Updated documentation for `PSP22Capped`.

### CODE REVIEW

File: docs/docs/smart-contracts/PSP22/Extensions/capped.md [hunk 1]
```diff
@@ -74,19 +74,19 @@ impl MyPSP22Capped {
...
         if (self.total_supply() + amount) > self.cap() {
-            return Err(PSP22Error::Custom(String::from("Cap exceeded")))
+            return Err(PSP22Error::Custom(String::from("Cap exceeded")));
         }
         PSP22Internal::_mint(self, account, amount)
     }
 
     /// Initializes the token's cap
     fn init_cap(&mut self, cap: Balance) -> Result<(), PSP22Error> {
         if cap <= 0 {
```
Comment (coreggon11) [inline docs/docs/smart-contracts/PSP22/Extensions/capped.md:88]: better to put this to the extension

File: tests/psp35_metadata.rs [hunk 2]
```diff
@@ -52,10 +52,10 @@ mod psp35_metadata {
     impl PSP35Metadata for PSP35Struct {}
 
     impl PSP35MetadataInternal for PSP35Struct {
-        fn _emit_attribute_set_event(&self, _id: &Id, _key: &Vec<u8>, _data: &Vec<u8>) {
+        fn _emit_attribute_set_event(&self, _id: &Id, _key: &[u8], _data: &Vec<u8>) {
             self.env().emit_event(AttributeSet {
                 id: _id.clone(),
-                key: _key.clone(),
+                key: _key.to_vec().clone(),
```
Comment (coreggon11) [inline tests/psp35_metadata.rs:58]: why can the parameter not be Vec?

File: contracts/src/token/psp34/extensions/metadata.rs [hunk 3]
```diff
@@ -60,12 +60,12 @@ impl<T: PSP34MetadataStorage<Data = PSP34MetadataData>> PSP34Metadata for T {
 }
 
 pub trait PSP34MetadataInternal {
-    fn _set_attribute(&mut self, id: Id, key: Vec<u8>, value: Vec<u8>);
+    fn _set_attribute(&mut self, id: Id, key: &[u8], value: Vec<u8>);
```
[the remark this answers is not in the document]
Comment (xgreenx) [reply contracts/src/token/psp34/extensions/metadata.rs:63]: We can't use slices in events=( Let's that remove that part of the change=) We will try to do that with new events

File: contracts/src/token/psp22/extensions/capped.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+declare_storage_trait!(PSP22CappedStorage);
+
+impl<T: PSP22CappedStorage<Data = PSP22CappedData> + PSP22 + PSP22Internal + InkStorage> PSP22Capped for T {
+    default fn cap(&self) -> Balance {
+        self.get().cap
+    }
+
+    default fn mint(&mut self, account: AccountId, amount: Balance) -> Result<(), PSP22Error> {
```
Comment (xgreenx) [inline contracts/src/token/psp22/extensions/capped.rs:53]: You don't need to add a new `mint` method=) You need to override the old ones. But it also is a bad idea. Better to have an internal method that does the check and if someone wants a `capped` restriction he should call that method. That method can get `from` and `value` and do the check

**Review by coreggon11 — APPROVED** [review]

File: docs/docs/smart-contracts/PSP22/Extensions/capped.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
+---
+sidebar_position: 7
+title: PSP22 Capped
+---
+
+This example shows how you can implement a [PSP22](https://github.com/Supercolony-net/openbrush-contracts/tree/main/contracts/src/token/psp22) contract with a supply cap, analogue to [ERC20Capped](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Capped.sol).
+
+## Step 1: Include dependencies
+
+Include `openbrush` as dependency in the cargo file or you can use [default `Cargo.toml`](/smart-contracts/overview#the-default-toml-of-your-project-with-openbrush) template.
```
Comment (xgreenx) [inline docs/docs/smart-contracts/PSP22/Extensions/capped.md:10]: We've changed the template in the ocuemntation. COuld you check for example `PSP22Metadata` doc?

File: examples/psp22_extensions/capped/lib.rs [hunk 6]
```diff
@@ -15,12 +18,17 @@ pub mod my_psp22_capped {
     pub struct Contract {
         #[storage_field]
         psp22: psp22::Data,
-        cap: Balance,
+        #[storage_field]
+        cap: Data,
```
Comment (xgreenx) [inline examples/psp22_extensions/capped/lib.rs:22]:
```suggestion
        cap: capped::Data,
```

File: contracts/src/token/psp22/extensions/capped.rs [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        self.data().cap.clone()
+    }
+}
+
+pub trait Internal {
+    /// Initializes the token's cap
+    fn _init_cap(&mut self, cap: Balance) -> Result<(), PSP22Error>;
+}
+
+impl<T: Storage<Data> + Storage<psp22::Data>> Internal for T {
```
Comment (xgreenx) [inline contracts/src/token/psp22/extensions/capped.rs:61]:
```suggestion
impl<T: Storage<Data>> Internal for T {
```

**Pull request merged**